### DISCUSSION TITLE
Seemingly random frame lag spikes during gameplay

### DISCUSSION THREAD

**Discussion by SupDos** [ppy/osu#17139] [Q&A] [answered]
For a while now I've been getting random lag spikes on the Draw thread (is that the right term?) every few minutes, which looks like this
![image](https://user-images.githubusercontent.com/6813986/156965920-dee205f1-311b-4328-abc7-f6605a65f9be.png) (big red spike)
on the advanced framerate overlay.

I have a couple recordings I made of it, and also have logs attached where I open the game, play until I get the lag spike and then quit in case it helps

https://user-images.githubusercontent.com/6813986/156966174-7b82a6b5-1e21-4e4d-89ce-7d8d070ec1e4.mp4

https://user-images.githubusercontent.com/6813986/156966187-04cc19e0-70c1-4afb-84a0-b6b7280871e8.mp4

https://user-images.githubusercontent.com/6813986/156966311-44fe05a6-e6d1-4f63-bcad-b8e7d88bda4b.mp4

Also tried with the default skin and same thing happens

https://user-images.githubusercontent.com/6813986/156966316-f3340dcf-ffc9-49bc-b184-8d10dab60757.mp4

[logs.zip](https://github.com/ppy/osu/files/8194726/logs.zip)

I've looked around and found #11800, would this be related to it? This only seems to occur during gameplay for me, sitting in the Main Menu has no stutters

**Comment by peppy** [comment, marked as answer]
From the frame graphs, this doesn't look GC related. It looks like a GPU pipeline stall (a lot of time spent in glSwap). Can you make sure you don't have other programs running in the background that could be interacting with your GPU and see if that helps?

**Reply by SupDos** [reply]
That pointed me in the right direction, with everything possible closed the System process in task manager spikes up to more than 10% GPU usage close to or exactly when the stutter happens so I'll have to look into it more on my side. Thanks!

**Reply by peppy** [reply]
Please let us know when you figure out what the cause is. Keep in mind that even if it's listed as `System`, it may still be another application which is invoking load on the DWM (I've seen this happen first hand).

**Reply by SupDos** [reply]
Small update on this, it's fixed! Seems like it was a GPU driver issue - finally decided to update them today after not doing so for a long time, but tried lazer right before updating them and it still had these stutters
Right after the update they are gone. Tested multiple maps including the ones above and no stutters whatsoever

**Reply by peppy** [reply]
Great to hear!

**Comment by Shirou27** [comment]
Lag spike is also happen in my phone(Redmi Note 9) but if i restart my phone before playing it doesn't.

**Comment by Flutterish** [comment]
This might not be particularly helpful, but lag spikes on the draw thread can sometimes happen if the shader stack is not empty at the end of a frame, although I think that is a framework check that only happens with the debugger attached.
